### DISCUSSION TITLE
Move C# to GDExtension,? C++ is for long? Python replace GDScript？and all in C++ ?

### DISCUSSION THREAD

**Discussion by wxlv97** [godotengine/godot-proposals#8630] [Scripting]
1、Will C# be considered for inclusion in GDExtension in the next release (5.0)? At that point, it will be on the same footing as C++, just like Rust. Just keep GDScript as the only scripting language for rapid development, as visual blueprints are for UE5. This will make the structure clearer and clearer.
2、 As a big fan of C++, I am eager to know if C++ will be supported by Godot for the long term, or if it will be removed, since not many people are developing Godot in C++, due to the well-known complexity of the language itself.
3、If Python replaces GDScript, I think it would be better because Python has better versatility. In another discussion, I learned that GDScript also calls C++ modules most of the time.
4、Since there are not many cases about C++ on the Internet(Youtube or BiliBili), I wonder if it is possible to develop a complete game in C++? Not even GDScript.

**Comment by AThousandShips** [comment]
Please do not make troll/joke discussions like this (assuming you're not laughing at your own post by accident), these points are all raised in other discussions/proposals.
